### ISSUE TITLE
Mod using too many ticks on dedicated server. 1.17.1

### ISSUE THREAD

**Issue by ItzNatsu** [lebonq/Automatic-Path#37]
I installed the mod on a server of mine because I thought it would bring more life to my modded server. everything was fine at first, and that probably was because we hadn't walked on that many grass blocks yet. after a few days however, the server started experiencing lots and lots of problems and lag. I was forced into installing a mod called Observable, it basically allows me to see what is causing lots of lag on the server. This mod was doing almost 7 million ticks. I assume that was every second. i was forced to remove the mod in order to clear the lag on my server. my server was running with 8GBs of ram. and it was on a separate device that had nothing running in the background with good connection. perfect for a dedicated server.

I think mobs being able to produce paths is part of the issue. because mobs walking everywhere creating path blocks would definitely cause some performance issues as it will be detecting these movements. Therefore, I think that if you attempt to optimize this mod. Allowing to toggle Mob path off through config would be great. 

On this screen shot, you can see just how much this mod ticks, and in the background, all those green highlighted blocks are path blocks. each of them individually ticking.
![unknown (1)](https://user-images.githubusercontent.com/68730384/136672995-ee4cb459-9cb2-49a2-b40c-63d153f91acd.png)

**Comment by lebonq** [comment]
Yes it's indeed a lot of ticks. I put it on the to-do list, first I will add a toggle switch for the creation of path by the mob. Then I'll look deeper in the mod to optimize it

**Comment by ItzNatsu** [comment]
I think one of the issues is that path blocks that aren't rendered are still ticking. Because using Observable I could detect path blocks far in the distance. That chunk wasn't even loaded I think. One thing I also think would be helpful is if I get to decide specifically how much each path block stage takes to be walked on to go up or down. Because then I could turn off mob paths, and then set the first stage to turn back to normal grass after a very short period of time. Which means that the only actual paths would be the ones we constantly walk on, and not the ones that we make while we go out adventuring.

**Comment by lebonq** [comment]
So I added a value to enable mob path creation. You can already tweak time between states of path block.
I don't close this issue because I think I can go deeper in the optimisation.

**Comment by ItzNatsu** [comment]
I think that you should also add a whitelist feature. Being able to select individual mobs to create paths would be great! Optimization + Customization. I'd probably use it to make it so that the only other entity that makes paths are villagers.

**Comment by josef-dijon** [comment]
What are you using to show that aggregated results list of us/t per block type?

**Comment by ItzNatsu** [comment]
> 

Observable. I don't remember whether it's on forge or fabric, maybe both. I'll leave that to you :)

**Comment by josef-dijon** [comment]
It's on both, thanks!

**Comment by UndoZan** [comment]
Has this changed at all?
